### ISSUE TITLE
Can't export a function out of a lib folder

### ISSUE THREAD

**Issue by MarkKahn** [facebook/flow#5996]
I have some opaque types that I want to be able to cast to through a single choke-point (for instance a userID, constrained number, date string, etc):

##### /type.js
```js
export opaque type specialType: string = string;

export function toSpecialType(str: string): specialType {
  // whatever processing
}
```

Now I want to use this opaque type in a libdef but nothing I try is working.

### Attempt 1

It doesn't appear that I can bring a type from outside of `/flow-typed` into it:

##### /flow-typed/lib.js
```js
import { specialType } from '../type.js'; // works

declare module 'foo' {
   declare function bar(): specialType;
}
```

##### /file.js
```js
import foo from 'foo';

const whatever = foo.bar(); // whatever = any
```

### Attempt 2

I move the contents of `type.js` into `/flow-typed`, but I can't require anything from `/flow-typed`:

##### /file.js
```js
import { toSpecialType } from './flow-typed';
```

this throws one of:

```
[flow] Cannot resolve module `/path/to/flow-typed/index.js`.
```
```
[flow] Internal error: package "<<PROJECT_ROOT>>/flow-typed/package.json" was not found in the PackageHeap!
```

depending on how I try to include it

For either case:

##### .flowconfig
```
[libs]
flow-typed
```

**Comment by MacMcIrish** [comment]
This also happens for custom defined lib folders, ie:
custom-types/myType.js
```
/* @flow */
export type myType = {...}
```
src/example.js
```
/* @flow */
import { myType } from './../custom-types/myType';
```
When my `.flowconfig` contains the following:
```
[lib]
./custom-types/
```
Then the resulting error is:
```
Cannot resolve module `/home/jeff/WebstormProjects/service-acl/custom-types/myType.js`.

   2| import type { myType } from "../../custom-types/myType";
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Removing `./custom-types/` from the `[lib]` section fixes the issue.

So it appears that direct exports from / imports from lib folders for types, not just functions, does not work.

**Comment by gnprice** [comment]
I think this issue can be closed, because the bug here was fixed in v0.143.0. There's also a documentation issue to be had in this area, but that's best treated as a separate thread.

The bug is that this:

> ##### /flow-typed/lib.js
> ```js
> import { specialType } from '../type.js'; // works
> ```

never actually worked, but before Flow v0.143.0 it resulted in no error message at the point of occurrence, and instead caused a set of very confusing symptoms including types turning to `any`. Since v0.143.0, Flow gives a nice clear error message: the import needs to get put inside a `declare module` block. See #6414 for details.

---

The documentation issue is that the docs don't clearly say that this isn't how you're supposed to import things defined in a libdef:

> I move the contents of `type.js` into `/flow-typed`, but I can't require anything from `/flow-typed`:
> 
> ##### /file.js
> ```js
> import { toSpecialType } from './flow-typed';
> ```

The way libdefs work, you never import from them by the pathname of the libdef file; rather, they have `declare module` blocks that refer to modules by name, and then you import from those modules by name. (Or they have declarations at the file toplevel, and then you don't import those at all -- they're present globally across your codebase. This is meant for things that are built into JS, or the browser or Node APIs, and so really are global.)

Here's the docs on libdefs:
https://flow.org/en/docs/libdefs/
https://flow.org/en/docs/libdefs/creation/
You can sort of get this from this line:
> The name specified in quotes after `declare module` can be any string, but it should correspond to the same string you’d use to `require` or `import` the third-party module into your project.

but it's sort of indirect and it'd be good to clarify explicitly what you are supposed to do, because importing them by pathname is a pretty natural thing to imagine as how they might work.

In any case, that documentation issue is best treated in a separate thread. The bug underlying this issue has been fixed, and the issue can be closed.
